src/test/java/com/code_intelligence/jazzer/mutation/mutator/StressTest.java

```diff
@@ -456,6 +456,7 @@ void genericMutatorStressTest(AnnotatedType type, String mutatorTree,
       testReadWriteExclusiveRoundtrip(mutator, fixedValue);
 
       initValues.add(mutator.detach(value));
```

Comment:
Instead of changing `value`, could we instead pass all the arguments to asserts through `fixFloatingPointsForProto`? That could have made this issue more obvious and also ensures that we don't interfere with the mutator - it should be able to assume that it is only passed values that it created itself.

Reply:
One problem is that if `value` is `some_field = -0.0`, then mutation of the underlying float/double value starts from `0.0`, so the mutator changes `value` to 0.0 when it converts the proto into float (or double). This seems to happen only because of the generic `Message:toBuilder()` function. Going the usual route `((FloatField3)value).getSomeField()` returns -0.0.
Our modification of `value` in the test mirrors this  proto-related modification and changes it to `some_field = 0.0` (or actually empty string because it's the default value).

The other problem is that when `detachedOldValue` is 0.0 and the mutated `value` is -0.0, `fixFloatingPointsForProto` changes `value` to 0.0 and the assertion fails as well.

Any ideas?

Reply:
Actually no ;-) This bug is plain terrible to work around. I will ping upstream and we can leave this as is.